File: Chapter 3/3.2.3 - nCr.cpp
#include<bits/stdc++.h>
using namespace std;
int nCr(int n, int r) {
    unsigned long long u = 1, d = 1, g;
    if (r > n - r) r = n - r;
    while (r) {
        u *= n--;
        d *= r--;
        g = __gcd(u, d);
        u /= g;
        d /= g;
    }
    return u / d;
}
int main() {
    cout << nCr(3, 2) << endl;
    return 0;
}

File: Chapter 3/3.2.3 - nCr.py
import math
def nCr(n, r):
    u = d = 1
    if(r > n - r):
        r = n - r
    while(r):
        u *= n; d *= r
        g = math.gcd(u, d)
        u /= g; d /= g
        n -= 1; r -= 1
    return u/d


n, r = map(int, input().split())
print(int(nCr(n, r)))